service/history/workflow/mutable_state_impl.go

```diff
@@ -5907,6 +5907,39 @@ func (ms *MutableStateImpl) closeTransactionTrackTombstones(
 	}
 
 	var tombstones []*persistencespb.StateMachineTombstone
+	if ms.stateMachineNode != nil {
+		opLog, err := ms.stateMachineNode.Outputs()
+		if err != nil {
+			panic(fmt.Sprintf("Failed to get HSM operation log: %v", err))
```

Comment:
Don't panic here, return an error instead and handle it in the caller.

Reply:
I'd be okay panicking if the only reason this would happen only unexpectedly. Consider making the `Outputs()` function panic instead of returning an error.